### ISSUE TITLE
Deterministic Variance Evaluation

### ISSUE THREAD

**Issue by codestuff001** [Zyin055/Inspect-Embedding-Training#20]
Would it be possible to adapt the embedding inspector to utilize this technique?

_Is This Loss Informative? Speeding Up Textual Inversion with Deterministic Objective Evaluation_
[https://arxiv.org/abs/2302.04841](https://arxiv.org/abs/2302.04841)

They come to a similar conclusion that training loss does not give any meaningful indication of training goals and describe an alternative method.

**Comment by Zyin055** [comment]
I didn't read the [12 page paper](https://arxiv.org/pdf/2302.04841.pdf), just the summary. Sounds like its just taking an average of the loss rate and stopping when it stops going down?

**Comment by codestuff001** [comment]
At the end of the paper they link to a github which has some examples:
https://github.com/yandex-research/DVAR

I don't know how it works, but I'm hoping there's some way to translate it into a graph that will better help determine when is a reliable stopping point during training.

**Comment by Zyin055** [comment]
Wouldn't the ideal solution be to just use their code that already does that?

**Comment by codestuff001** [comment]
Yes, that would be great, but I'm unfamiliar with how to use it. I currently use Automatic1111 to train textual inversion, how would I use the code to evaluate an embedding that I've trained with the webui?

I see this command line argument on the github page:
```
export CUDA_VISIBLE_DEVICES="your-gpus"

python textual_inversion.py \
  -m [sd/ldm] \
  --train_data_dir="path-to-cat_statue-images" \
  --placeholder_token="inversion-cat_statue"
  --init_strategy best
  --train_batch_size=4 \
  --eval_batch_size=4 \
  --save_steps 100 \
  --max_train_steps 6100 \
  --learning_rate 5e-3 \
  --scale_lr \
  --sample_frequency -1 \
  --n_val_prompts 8 \
  --variant dvar_early_stopping \
  --early_stop_eps 0.39 \
  --early_stop_patience 282 \
  --mixed_precision fp16 \
  --logger wandb \
  --seed 59
```

As far as I understand it runs the textual inversion script. What if I just want to evaluate an embedding to generate a deterministic loss graph?
